src/SeleniumLibrary/keywords/selectelement.py

```diff
@@ -18,206 +18,196 @@
 
 from SeleniumLibrary.base import LibraryComponent, keyword
-from SeleniumLibrary.utils import is_truthy
+from SeleniumLibrary.utils import is_truthy, plural_or_not as s
 
 
 class SelectElementKeywords(LibraryComponent):
 
     @keyword
-    def get_list_items(self, locator, value=False):
-        """Returns the labels or values in the select list identified by `locator`.
+    def get_list_items(self, locator, values=False):
+        """Returns all labels or values of selection list ``locator``.
 
-        Select list keywords work on both lists and combo boxes. Key attributes for
-        select lists are `id` and `name`. See `introduction` for details about
-        locating elements.
+        See the `Locating elements` section for details about the locator
+        syntax.
+
+        Returns visible labels by default, but values can be returned by
+        setting the ``values`` argument to a true value (see `Boolean
+        arguments`).
 
         Example:
-        | ${labels1} = | Get List Items | xpath=//h1 |
-        | ${labels2} = | Get List Items | xpath=//h1 | value=${False} |
-        | ${values} = | Get List Items | xpath=//h1 | value=True |
-        | Should Be Equal | ${labels1} | ${labels2} |
+        | ${labels} = | Get List Items | mylist                |             |
+        | ${values} = | Get List Items | css:#container select | values=True |
+
+        Support to return values is new in SeleniumLibrary 3.0.
         """
-        select, options = self._get_select_list_options(locator)
-        if is_truthy(value):
-            return self._get_values_for_options(options)
+        options = self._get_options(locator)
+        if is_truthy(values):
+            return self._get_values(options)
         else:
-            return self._get_labels_for_options(options)
+            return self._get_labels(options)
 
     @keyword
     def get_selected_list_label(self, locator):
-        """Returns the visible label of the selected element from the select list identified by `locator`.
+        """Returns label of selected option from selection list ``locator``.
+
+        If there are multiple selected options, label of the first option
+        is returned.
 
-        Select list keywords work on both lists and combo boxes. Key attributes for
-        select lists are `id` and `name`. See `introduction` for details about
-        locating elements.
+        See the `Locating elements` section for details about the locator
+        syntax.
         """
         select = self._get_select_list(locator)
         return select.first_selected_option.text
 
     @keyword
     def get_selected_list_labels(self, locator):
-        """Returns the visible labels of selected elements (as a list) from the select list identified by `locator`.
+        """Returns labels of selected options from selection list ``locator``.
 
-        Fails if there is no selection.
+        Starting from SeleniumLibrary 3.0, returns an empty list if there
```

Comment:
Should this chapter be last lines of the doc string? So that in future they are easy to spot and remove?

Reply:
In my opinion it's fine either way. It's anyway easy to find all references to SeleniumLibrary 3.0 and removing this note isn't hard either. The main reason I decided not to have the note at the end was that we'd end up with two very short paragraphs. If you want to change it, go ahead.

Reply:
Agreed